### PR TITLE
Simplify fields combination validator interface

### PR DESCRIPTION
**Pull request by dimagorelik**
Simplify fields combination validator interface

### CODE REVIEW

File: main/src/main/java/com/kenshoo/pl/entity/spi/helpers/FieldsCombinationCreator.java [hunk 1]
```diff
@@ -15,7 +15,7 @@ public class FieldsCombinationCreator {
 
     public static <E extends EntityType<E>> Collection<? extends FieldsValueMap<E>> create(final Collection<? extends EntityChange<E>> entityChanges, final Collection<EntityField<E, ?>> fields, ChangeOperation changeOperation, ChangeContext changeContext) {
```
Comment (dimagorelik) [inline main/src/main/java/com/kenshoo/pl/entity/spi/helpers/FieldsCombinationCreator.java:16]: @effiban 
This utility class is not used by the framework. Maybe we can remove it from here and recreate it in the client code?

File: main/src/main/java/com/kenshoo/pl/entity/internal/FieldsCombination.java [hunk 2]
```diff
@@ -12,13 +12,18 @@
 
 import static java.util.stream.Collectors.toSet;
 
-public class ResultingFieldsCombination<E extends EntityType<E>> implements FieldsValueMap<E> {
+/**
+ * The final state of fields in the database.
```
Comment (effiban) [inline main/src/main/java/com/kenshoo/pl/entity/internal/FieldsCombination.java:16]:
```suggestion
 * The final state of a certain combination of fields in the database.
```

File: main/src/main/java/com/kenshoo/pl/entity/internal/FieldsCombination.java [hunk 3]
```diff
@@ -12,13 +12,18 @@
...
+ * This will be the state of fields after all changes have been applied on the current state and contains the final values of the entity's fields.
+ *
+ */
+public class FieldsCombination<E extends EntityType<E>> implements FieldsValueMap<E> {
```
Comment (effiban) [inline main/src/main/java/com/kenshoo/pl/entity/internal/FieldsCombination.java:20]: How about `FinalFieldsCombination` ?

File: main/src/main/java/com/kenshoo/pl/entity/spi/helpers/FieldsCombinationCreator.java (hunk 1 above)
Comment (effiban) [reply main/src/main/java/com/kenshoo/pl/entity/spi/helpers/FieldsCombinationCreator.java:16]: @dimagorelik - not sure, it looks like this is a helper that makes things easier for the client so we wouldn't want to move it there - but I haven't looked at the usages yet.

File: main/src/main/java/com/kenshoo/pl/entity/internal/FieldsCombination.java (hunk 3 above)
Comment (dimagorelik) [reply main/src/main/java/com/kenshoo/pl/entity/internal/FieldsCombination.java:20]: @effiban 
In this case, I prefer FieldsCombination

**Review by effiban — APPROVED** [review]

**Pull request merged**